custom_components/versatile_thermostat/thermostat_switch.py

```diff
@@ -101,6 +102,7 @@ def post_init(self, config_entry):
                     thermostat=self,
                     switch_entity_id=switch,
                     initial_delay_sec=idx * delta_cycle,
+                    keep_alive_sec=config_entry.get(CONF_HEATER_KEEP_ALIVE),
```

Comment:
Actually, if it is just `None` that we are guarding against, the following should be enough, no?
```python
UnderlyingSwitch(
    ... ,
    keep_alive_sec=config_entry.get(CONF_HEATER_KEEP_ALIVE, 0),
)
```
I did some testing and I can confirm that:
* `config_entry.get(SOME_UNSET_ENTRY, 0)` returns `0` rather than `None`.
* When using the Home Assistant integration configuration user interface (config flow), the UI prevents me from entering negative numbers and decimal points, and if I delete the contents of the form field (unsetting it), `config_entry.get(SOME_UNSET_ENTRY)` actually returns `0` rather than `None`, thanks to the schema constraint in `config_schema.py`:

```python
vol.Optional(CONF_HEATER_KEEP_ALIVE, default=0): cv.positive_int,
```

Reply:
I would do that:
```
keep_alive_sec=config_entry.get(CONF_HEATER_KEEP_ALIVE) or 0 
```
but 
```
config_entry.get(CONF_HEATER_KEEP_ALIVE, 0)
```
(which I don't was aware until now), seems good also.

Reply:
`config_entry` has the type of [ConfigEntry.data](https://github.com/home-assistant/core/blob/2024.1.5/homeassistant/config_entries.py#L242), which is [collections.abc.Mapping](https://docs.python.org/3.11/library/collections.abc.html#collections.abc.Mapping), which is specified at:
https://github.com/python/cpython/blob/v3.11.7/Lib/_collections_abc.py#L771

While investigating this, I got a bit carried away and created a separate PR that adds more type hints in a few files: #364